qiskit/transpiler/passes/optimization/commutation_analysis.py

```diff
@@ -115,8 +116,16 @@ def _commute(node1, node2):
 
     id_op = Operator(np.eye(2 ** qbit_num))
 
-    op12 = id_op.compose(node1.op, qargs=qarg1).compose(node2.op, qargs=qarg2)
-    op21 = id_op.compose(node2.op, qargs=qarg2).compose(node1.op, qargs=qarg1)
+    if ((id(node1.op), str(qarg1)), (id(node2.op), str(qarg2))) in cache:
```

Comment:
Similar to the `id` for the op, `(type(qarg1), qarg1.register.name, qarg1.index)` might be quicker to generate and compare (but it's worth benchmarking.)